sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -58,6 +58,8 @@ public static ClientPipeline Create(ClientPipelineOptions? options = default)
     {
         Argument.AssertNotNull(options, nameof(options));
 
+        options.Freeze();
```

Comment:
We freeze pipeline options automatically when a pipeline is constructed in ClientModel-based libraries.  Making the Freeze method public and explicit means Azure.Core clients don't get this behavior automatically in ClientOptions, which will inherit from ClientPipelineOptions in Azure.Core 2.0.